nodeInterface/NodeInterface.go

```diff
@@ -590,3 +590,83 @@ func (n NodeInterface) LegacyLookupMessageBatchProof(c ctx, evm mech, batchNum h
 	calldataForL1 = data
 	return
 }
+
+func (n NodeInterface) getL1BlockNum(l2BlockNum uint64) (uint64, error) {
+	blockHeader, err := n.backend.HeaderByNumber(n.context, rpc.BlockNumber(l2BlockNum))
+	if err != nil {
+		return 0, err
+	}
+	l1BlockNum := types.DeserializeHeaderExtraInformation(blockHeader).L1BlockNumber
+	return l1BlockNum, nil
+}
+
+func (n NodeInterface) GetL2BlockRangeForL1(c ctx, evm mech, l1BlockNum uint64) ([]uint64, error) {
```

Comment:
We should return range (e.g. uint64, uint64 or define struct with two ends, etc..) not arbitrary sized slice.